configure.ac

```diff
@@ -18,6 +18,16 @@ PKG_PROG_PKG_CONFIG([])
 
 AC_CHECK_HEADERS([sys/capability.h], [], [AC_MSG_ERROR([*** POSIX caps headers not found])])
 
+AC_ARG_ENABLE(warn_implicit_session_opt,
+              [AS_HELP_STRING([--enable-warn-implicit-session-opt],
+                              [warn when the new terminal session creation behaviour isn't specified explicitly [default=no]])],
+              [warn_implicit_session_opt=yes],
+              [warn_implicit_session_opt=no])
+
+AS_IF([test "$warn_implicit_session_opt" == yes], [
+  AC_DEFINE([WARN_IMPLICIT_SESSION_OPT], [], [Warn when the new terminal session creation behaviour isn't specified explicitly])
+])
+
```

Comment:
This amount of boilerplate makes me increasingly tempted to delete the Autotools build system so we're not maintaining two in parallel.